Workaround for handling private endpoints on KeyVault creation

theccz opened the pull request:
## Description

When trying to create a keyvault with a key in the same terraform run the run fails as the private endpoint needs to create first before a key is created. Leveraging azapi ensures the key is created in the same run.

<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Closes #456
-->

## Type of Change

<!-- Use the check-boxes [x] on the options that are relevant. -->

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates.
  - [x] Breaking changes.
  - [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/terraform -->

matt-FFFFFF commented:
This is a fundamental and breaking change to the module. Is there another way to achieve what you need?

theccz commented:
In the end we achieved what we wanted using the enable public network which creates a public IP on endpoint creation, then on subsequent runs the endpoint sits on the private endpoint.

Before this we tried different ways to take ownership of the key/secret with the azapi resource, however we were not successful. Some of the options we tried were playing with lifecycle and destroyed meta-arguments. As well as trying using the moved argument:

```
moved {
  from = azurerm_key_vault_secret.this
  to = azapi_resource.secret
}
```
But azapi_resource does not support it.

```
│ Error: Unsupported `moved` across resource types
│
│   on ../../modules/key/[main.tf](http://main.tf/) line 30:
│   30: moved {
│
│ The provider "[registry.terraform.io/azure/azapi](http://registry.terraform.io/azure/azapi)" does not support moved operations across resource types and providers.
╵
╷
│ Error: Unsupported `moved` across resource types
│
│   on ../../modules/secret/[main.tf](http://main.tf/) line 20:
│   20: moved {
│
│ The provider "[registry.terraform.io/azure/azapi](http://registry.terraform.io/azure/azapi)" does not support moved operations across resource types and providers.
```
